controllers/snow_machineconfig_controller.go

```diff
@@ -19,18 +19,24 @@ type ClientBuilder interface {
 	BuildSnowAwsClientMap(ctx context.Context) (aws.Clients, error)
 }
 
+type ValidatorBuilder interface {
+	Build(aws aws.Clients) snow.Validator
```

Comment:
This would require a lot more refactoring (I think in `entry.go`) from the cli workflow side too because right now we don't need to maintain context of the aws clients I think. Yea if we can avoid the builder that is probably best, but maybe we have a better way to solve this once we change how we use validator for both workflows?